_inc/connect-button.js

```diff
@@ -80,12 +80,20 @@ jQuery( document ).ready( function( $ ) {
 		},
 		receiveData: function( event ) {
 			if (
-				event.origin === jpConnect.jetpackApiDomain &&
-				event.source === jetpackConnectIframe.get( 0 ).contentWindow &&
-				event.data === 'close'
+				event.origin !== jpConnect.jetpackApiDomain ||
+				event.source !== jetpackConnectIframe.get( 0 ).contentWindow
 			) {
-				window.removeEventListener( 'message', this.receiveData );
-				jetpackConnectButton.handleAuthorizationComplete();
+				return;
+			}
+
+			var type = event.data.type || 'close';
```

Comment:
I'm not sure why we should default to `close` here. I think it's odd to allow for any `event.data.type` here. Should we just do nothing if the type isn't between the ones we expect?